controllers/knative/knative.go

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package knative
+
+import (
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+	clienteventingv1 "knative.dev/eventing/pkg/client/clientset/versioned/typed/eventing/v1"
+	clientservingv1 "knative.dev/serving/pkg/client/clientset/versioned/typed/serving/v1"
+)
+
+var servingClient clientservingv1.ServingV1Interface
+var eventingClient clienteventingv1.EventingV1Interface
+
+type Info struct {
+	IsKnativeEventing bool
+	IsKnativeServing  bool
+}
```

Comment:
```suggestion
type Availability struct {
	Eventing bool
	Serving  bool
}
```

So we have from external packages: `knative.GetAvailability().Serving`

We avoid repetition and `Info` is too generic. wdyt?